### PR TITLE
pass initial TLS certificates into RSS

### LINKED ISSUE #1528: RSS -> Nexus handoff plumbing
RSS must provide operators a way to "add sufficient networking configuration to be able to launch Nexus, and continue setup through the web console".

The current implementation is somewhat limited in this regard - although `config-rss.toml` - representing user input - can contain an external address for Nexus, there are shortcomings:
- It's limited to a single address
- There is no mechanism to supply server certificates to encrypt traffic

### PR DESCRIPTION
**Pull request by davepacheco**
Right now, RSS always provides an empty list of initial TLS certificates (for the external API) to Nexus during rack initialization.  On a real system, the user will provide the initial certificate over the technician port during initial setup (see RFDs 57, 323).  This means Wicket will get them, and it will need to provide them to RSS.  This PR adds these as a parameter to RSS's rack-initialize endpoint.

This affects different environments:

- In a "real" system (which is just dogfood right now), this will need to be provided by whoever's crafting the initial RSS request.  For Dogfood, we'll want to take the existing certificate and key and put them in PEM format into the initial RSS request.
- In the CI helios-deploy job that runs end-to-end tests, in this PR we generate a self-signed certificate, have RSS get that, and have the end-to-end tests reqwest client explicitly trust this certificate.
- In other non-simulated dev environments (i.e., if you deploy on your own using the how-to-run instructions), if you do nothing, you will have no certificates.  (No change from today.)  Nexus will only listen over HTTP.  You _can_ provide certificate(s), in which case Nexus will listen over HTTPS, too.
- In simulated environments (including the test suite), you will have no certificates.  (No change from today.)

Fixes #1528.

This also addresses one of the two remaining items in #1529.  The other one is notifying Nexus instances when one of them receives a certificate update.  In an upcoming PR, I intend to instead have Nexus instances have a background task to monitor the list of certs.

### CODE REVIEW

**Review by luqmana — COMMENTED** [review]
a few minor comments but looks good thanks. i would update the how-to-run doc at least to make mention of this

File: smf/sled-agent/gimlet-standalone/config-rss.toml [hunk 1]
```diff
@@ -20,6 +20,9 @@ dns_servers = [ "1.1.1.1", "9.9.9.9" ]
 # Delegated external DNS zone name
 external_dns_zone_name = "oxide.test"
 
+# Initial TLS certificates for the external API
+external_certificates = []
```
[the remark this answers is not in the document]
Comment (davepacheco) [reply smf/sled-agent/gimlet-standalone/config-rss.toml:34]: Yeah, good idea.  I'm a little worried about this getting out of date though.  Instead I added a note both here and at the top of the file about where to find the actual definition.  How's that?

File: sled-agent/src/rack_setup/config.rs [hunk 2]
```diff
@@ -110,4 +161,68 @@ mod test {
...
+                    domain, error
+                )
+            });
+
+        // Write the configuration file.
+        let cfg_bytes = std::fs::read(&path).unwrap();
+        let cfg_path = tempdir.path().join("config-rss.toml");
+        std::fs::write(&cfg_path, &cfg_bytes)
+            .with_context(|| format!("failed to write to {:?}", &tempdir))
+            .unwrap();
```
[the remark this answers is not in the document]
Comment (davepacheco) [reply sled-agent/src/rack_setup/config.rs]: Done in 57b9eec4ffac098bc892589648ed1a5f08b8dc5d.

**Comment by davepacheco** [comment]
Thanks for the review!  I think I addressed the feedback in 57b9eec4ffac098bc892589648ed1a5f08b8dc5d.

**Review by luqmana — APPROVED** [review]
Thanks!

**Pull request merged**